docs(design): add latency definition, records service and visualization (#106)
tier4/CARET_doc: docs/design/latency_definitions/path.md

# Path

> Path latency is the sum of node latency and communication latency included in the path.

Reviewer: ```suggestion
Path latency is sum of node latency and communication latency included in a certain path.
```

> Path latency is sum of node latency and communication latency included in a certain path.

Reviewer: ```suggestion
Path latency is sum of node latency and communication latency included in a certain path. Path represents data flow which  is constructed of multiple nodes connected each other. 
```

Author: [2470a77](https://github.com/tier4/CARET_doc/pull/106/commits/2470a77b1216671bd87d9ba48d765b935799ac11)

$$
l_{path} = \sum_{\in path} l_{node} + \sum_{\in path} l_{comm} \\
l_{node} = t_{pub} - t_{sub} \\
l_{comm} = t_{sub} - t_{pub} \\
$$

> Here, path is one of the paths chosen from the node graph.

Reviewer: Could you tell me how `the node graph` is related to `the paths`?

Reviewer: Thanks, but I'd like to improve this sentence at next comment.

> Here, path is one of the paths chosen from the node graph.

Reviewer: ```suggestion
```

Reviewer: Moved https://github.com/tier4/CARET_doc/pull/106/files#r1031346098

Author: [2470a77](https://github.com/tier4/CARET_doc/pull/106/commits/2470a77b1216671bd87d9ba48d765b935799ac11)

In CARET, the path is defined as `[node_name]-[topic_name]-... -[topic_name]-[node_name]`.  
For example, in the following case, the path definition is `[A]-[/a]-[C]-[/e]-[E]`.

![node path](../../imgs/node_path.drawio.png)

<prettier-ignore-start>
!!! Info

>     In the above definition, the start time of the latency is the start node's Publish and the end time of the latency measurement is the end node's Subscribe.

Reviewer: ```suggestion
    In the above definition, the starting time of the latency is publish of the starting node and the end time of the latency is subscription of the end node.
```

>     In the above definition, the starting time of the latency is publish of the initial node and the end time of the latency is subscription of the last node.

Reviewer: Sorry, but I took a mistake.

```suggestion
    In the above definition, the star time of the latency is publish of the initial node and the end time of the latency is subscription of the last node.
```
